nsv13/code/controllers/subsystem/starsystem.dm

```diff
@@ -0,0 +1,206 @@
+			priority_announce("[message]", "Naval Command") //Warn players for idleing too long
+		if(modifier == 22) // 45 minutes
+			var/total_deductions
+			for(var/account in SSeconomy.department_accounts)
+				var/datum/bank_account/D = SSeconomy.get_dep_account(account)
+				total_deductions += D.account_balance / 2
+				D.account_balance = D.account_balance / 2
+			priority_announce("Significant damage has been caused to NanoTrasen assets due to the inactivity of your vessel. [total_deductions] credits have been deducted across all departmental budgets to cover expenses.")
+		if(istype(LH))
+			LH.weight += 1 //Increment probabilty via SSEvent
+			if(LH.weight % 5 == 0)
+				message_admins("The ship has been out of combat for [last_combat_enter]. The weight of [LH.name] is now [LH.weight]")
+				log_game("The ship has been out of combat for [last_combat_enter]. The weight of [LH.name] is now [LH.weight]")
+		if(istype(BR))
+			BR.weight += 1 //Increment probabilty via SSEvent
+			if(BR.weight % 5 == 0)
+				message_admins("The ship has been out of combat for [last_combat_enter]. The weight of [BR.name] is now [BR.weight]")
+				log_game("The ship has been out of combat for [last_combat_enter]. The weight of [BR.name] is now [BR.weight]")
+
+/datum/controller/subsystem/starsystem/proc/event_info() //Admin command for debugging output
+	var/datum/round_event_control/lone_hunter/LH = locate(/datum/round_event_control/lone_hunter) in SSevents.control
+	var/datum/round_event_control/belt_rats/BR = locate(/datum/round_event_control/belt_rats) in SSevents.control
+	message_admins("Current time: [world.time] | Last Combat: [last_combat_enter] | Modifier: [modifier] | [LH.name]: [LH.weight] | [BR.name]: [BR.weight]")
+
+/datum/controller/subsystem/starsystem/proc/weighting_reset() //All overmap combat events need to be populated in this proc - this is used to reset the weight of every combat encounter once an encounter is spawned
+	var/datum/round_event_control/lone_hunter/LH = locate(/datum/round_event_control/lone_hunter) in SSevents.control
+	var/datum/round_event_control/belt_rats/BR = locate(/datum/round_event_control/belt_rats) in SSevents.control
+	LH.weight = 0
+	BR.weight = 0
+
+/datum/controller/subsystem/starsystem/New()
+	. = ..()
+	instantiate_systems()
+
+/datum/controller/subsystem/starsystem/proc/instantiate_systems()
+	cycle_gameplay_loop() //Start the gameplay loop
+	cycle_bounty_timer() //Start the bounty timers
+	var/datum/starsystem/SS = new
+	hyperspace = SS //First system defaults to "hyperspace" AKA transit.
+	for(var/instance in subtypesof(/datum/starsystem))
+		var/datum/starsystem/S = new instance
+		systems += S
+
+///////SPAWN SYSTEM///////
+
+/datum/controller/subsystem/starsystem/proc/find_main_overmap() //Find the main ship
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects)
+		if(OM.role == MAIN_OVERMAP)

```

Comment:
Remember to change the "main_overmap" vars on other ships and make a isMainOvermap() method to replace them (if not already done so)